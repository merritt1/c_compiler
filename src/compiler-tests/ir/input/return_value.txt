int functionName(void) {
	int j;
	j = 0;
	return j;
	j = 12;
}
void main(void) {
	int i;
	i = functionName();
}