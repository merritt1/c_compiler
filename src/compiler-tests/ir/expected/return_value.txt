(procBegin,  _Global_functionName)
(addressOf,     r0,      j)
(constInt,     r1,          0)
(storeWord,     r0,      r1)
(addressOf,     r2,      j)
(loadWord,     r3,      r2)
(returnWord,     r3)
(goto    ,  _GeneratedLabel_1)
(addressOf,     r4,      j)
(constInt,     r5,         12)
(storeWord,     r4,      r5)
(label   ,  _GeneratedLabel_1)
(procEnd ,  _Global_functionName)
(procBegin,  main)
(addressOf,     r6,      i)
(call    ,  _Global_functionName)
(resultWord,     r7)
(storeWord,     r6,      r7)
(procEnd ,  main)
